src/query/storage/m3/multiresults/multi_fetch_result.go

```diff
@@ -166,24 +169,24 @@ func (r *multiResult) addOrUpdateDedupeMap(
 	newIterators encoding.SeriesIterators,
 ) {
 	for _, iter := range newIterators.Iters() {
-		id := iter.ID().String()
+		id := iter.ID().Bytes()
 
-		existing, exists := r.dedupeMap[id]
+		existing, exists := r.dedupeMap.Get(id)
 		if !exists {
 			// Does not exist, new addition
-			r.dedupeMap[id] = multiResultSeries{
+			r.dedupeMap.Set(id, multiResultSeries{
```

Comment:
more a question: should this be `SetUnsafe`? i.e. can you rely on the bytes underlying `id` for the lifecycle you need in this request-response cycle?

Reply:
Good call, these are all for builders that are much shorter lived than the objects they're building